### ISSUE TITLE
extend media for cast an additional column on model

### ISSUE THREAD

**Issue by tkart38** [plank/laravel-mediable#192]
hi 
i've add a column on media table (custom_fields) in json format.
i would like to cast custom_fields but if i extend the media model the cast is not reconized.

i've add my new class in  config/mediable.php nothin happend.

how can i do?
thanks

**Comment by frasmage** [comment]
Hi @tkart38,

In order to help you debug, could you provide snippets of your `Media` subclass and the relevant lines of your `config/mediable.php` file?

**Comment by Webpills** [comment]
hi @frasmage,
this is Mymedia.php file in \app\Models ,
i need to use the $cast for custom_fiels (additional colum on Media table) and the softdelete function.
i think i'm missing something.
`<?php

namespace App\Models;

use Plank\Mediable\Mediable;
use Illuminate\Database\Eloquent\SoftDeletes;


class Mymedia extends Model {
  
    use SoftDeletes;

    protected $casts = [
        'custom_fields' => 'json',
    ];
    

}`
extract of config/mediable.php
/*
     * FQCN of the model to use for media
     *
     * Should extend `Plank\Mediable\Media`
     */
    'model' => Plank\Mediable\Media::class,
    'model' => App\Models\Mymedia::class,

thanks for your attention

**Comment by tkart38** [comment]
hi @frasmage,
this is Mymedia.php file in \app\Models ,
i need to use the $cast for custom_fiels (additional colum on Media table) and the softdelete function.
i think i'm missing something.
`<?php

namespace App\Models;

use Plank\Mediable\Mediable;
use Illuminate\Database\Eloquent\SoftDeletes;

class Mymedia extends Model {

use SoftDeletes;

protected $casts = [
    'custom_fields' => 'json',
];
}extract of config/mediable.php /* * FQCN of the model to use for media * * Should extendPlank\Mediable\Media`
*/
'model' => Plank\Mediable\Media::class,
'model' => App\Models\Mymedia::class,

thanks for your attention

**Comment by frasmage** [comment]
You `MyMedia` class needs to extend `Plank\Mediable\Media`, not your base Model class

**Comment by tkart38** [comment]
hi thanks for your help,
i'm doing this 

```
<?php

namespace App\Models;

use Plank\Mediable\Media;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Database\Eloquent\Model;


class Mymedia extends Media {

    public static function test(){
        return 'test';
    }
  
    use SoftDeletes;

    protected $casts = [
        'custom_fields' => 'json',
    ];
    }
```
this in config
```
   /*
     * FQCN of the model to use for media
     *
     * Should extend `Plank\Mediable\Media`
     */
  
    'model' => App\Models\Mymedia::class,
```

can use test on view with this:
 {{ App\Models\Mymedia::test() }}

but not like this:

$media->test()

and the soft delete does'nt work.

thanks in advances

**Comment by frasmage** [comment]
I'm not following, what does soft deletion have to do with a random static method?

As per [Laravel Docs](https://laravel.com/docs/4.2/eloquent#soft-deleting), soft deletion requires an additional `deleted_at` column, you will need to do that yourself.

**Comment by tkart38** [comment]
hi,
yes you are right, we have to think to one problem at a time.

### softdelete
i've added the colum `deleted_at` on media table, but need to "active" the softdelete.
i'm trying to extend media class for add `use SoftDeletes;` and use it.
now when i delete an image the row on Media table is deteted and not update on `deleted_at` column.

The static method is for other reason, ignore it now.

thanks again

**Comment by tkart38** [comment]
hi 
ok i solved my problem i'm getting the wrong model.
Thanks for the support

**Comment by tkart38** [comment]
Sorry but now i can't get the Media Attached on other Model.
I'm trying this $article->firstMedia('thumbnail');  
but give this error:
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'mediables.mymedia_id' in 'field list'.....

```
<?php

namespace App\Models;

use Plank\Mediable\Media;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Database\Eloquent\Model;


class Mymedia extends Media {

    protected $table = 'media';
    
    public static function test(){
        return 'test';
    }
  
    use SoftDeletes;

    protected $casts = [
        'custom_fields' => 'json',
    ];
    

}

```
I think it is generating the relation field automatically from the extended model name (Mymedia), while DB schema is the original  from your package.

what should i do?
thanks

**Comment by frasmage** [comment]
That's a bug, should be fixed in version 4.4.1
